assets/js/components/ExchangeRates.js

```diff
@@ -1,106 +1,175 @@
-// ./assets/js/components/ExchangeRates.js
-
-import React, {Component} from 'react';
+import React, { useState, useEffect } from 'react';
 import axios from 'axios';
 
-class ExchangeRates extends Component {
-    constructor() {
-        super();
-        this.state = {
-            rates: {},
-            loading: true,
-            date: (new Date()).toISOString().split('T')[0]
-        };
-    }
+const ExchangeRates = () => {
+    const [rates, setRates] = useState({});
+    const [loading, setLoading] = useState(true);
+    const [date, setDate] = useState((new Date()).toISOString().split('T')[0]);
+    const [calculateValue, setCalculateValue] = useState('');
+    const [calculateBuy, setCalculateBuy] = useState(-1);
+    const [calculateSell, setCalculateSell] = useState(-1);
+    const [calculateExchange, setCalculateExchange] = useState('');
+    const [calculateCode, setCalculateCode] = useState('');
```

Comment:
Tu mógłbyś użyć jednego const'a:
```suggestion
    const [rates, setRates] = useState({}),
        [loading, setLoading] = useState(true),
        [date, setDate] = useState((new Date()).toISOString().split('T')[0]),
        [calculateValue, setCalculateValue] = useState(''),
        [calculateBuy, setCalculateBuy] = useState(-1),
        [calculateSell, setCalculateSell] = useState(-1),
        [calculateExchange, setCalculateExchange] = useState(''),
        [calculateCode, setCalculateCode] = useState('')
    ;
```